x-pack/legacy/plugins/lens/public/editor_frame_service/editor_frame/config_panel_wrapper.tsx

```diff
@@ -174,85 +190,329 @@ function LayerPanels(
+    visualizationState: unknown;
+    updateVisualization: StateSetter<unknown>;
+    updateDatasource: (datasourceId: string, newState: unknown) => void;
+    onRemove: () => void;
+  }
 ) {
+  const dragDropContext = useContext(DragContext);
   const { framePublicAPI, layerId, activeVisualization, isOnlyLayer, onRemove } = props;
   const datasourcePublicAPI = framePublicAPI.datasourceLayers[layerId];
-  const layerConfigProps = {
+  if (!datasourcePublicAPI) {
+    return <></>;
+  }
+  const layerVisualizationConfigProps = {
     layerId,
-    dragDropContext: props.dragDropContext,
+    dragDropContext,
     state: props.visualizationState,
-    setState: props.setState,
+    setState: props.updateVisualization,
     frame: props.framePublicAPI,
+    dateRange: props.framePublicAPI.dateRange,
   };
+  const datasourceId = datasourcePublicAPI.datasourceId;
+  const layerDatasourceState = props.datasourceStates[datasourceId].state;
+  const layerDatasource = props.datasourceMap[datasourceId];
 
-  return (
-    <EuiPanel className="lnsConfigPanel__panel" paddingSize="s">
-      <EuiFlexGroup gutterSize="s" alignItems="flexStart" responsive={false}>
-        <EuiFlexItem grow={false}>
-          <LayerSettings
-            layerId={layerId}
-            layerConfigProps={layerConfigProps}
-            activeVisualization={activeVisualization}
-          />
-        </EuiFlexItem>
+  const layerDatasourceDropProps = {
+    layerId,
+    dragDropContext,
+    state: layerDatasourceState,
+    setState: (newState: unknown) => {
+      props.updateDatasource(datasourceId, newState);
+    },
+  };
+
+  const layerDatasourceConfigProps = {
+    ...layerDatasourceDropProps,
+    frame: props.framePublicAPI,
+    dateRange: props.framePublicAPI.dateRange,
+  };
 
-        {datasourcePublicAPI && (
-          <EuiFlexItem className="eui-textTruncate">
-            <NativeRenderer
-              render={datasourcePublicAPI.renderLayerPanel}
-              nativeProps={{ layerId }}
+  const [popoverState, setPopoverState] = useState<{
+    isOpen: boolean;
+    openId: string | null;
+  }>({
+    isOpen: false,
+    openId: null,
+  });
+
+  const { dimensions } = activeVisualization.getLayerOptions(layerVisualizationConfigProps);
+  const isEmptyLayer = !dimensions.some(d => d.accessors.length > 0);
+
+  function wrapInPopover(id: string, trigger: React.ReactElement, panel: React.ReactElement) {
+    return (
+      <EuiPopover
+        isOpen={popoverState.isOpen && popoverState.openId === id}
+        closePopover={() => {
+          setPopoverState({ isOpen: false, openId: null });
+        }}
+        button={trigger}
+        display="block"
+        anchorPosition="leftUp"
+        withTitle
+        panelPaddingSize="s"
+      >
+        {panel}
+      </EuiPopover>
+    );
+  }
+
+  return (
+    <ChildDragDropProvider {...dragDropContext}>
+      <EuiPanel className="lnsConfigPanel__panel" paddingSize="s">
```

Comment:
For reviewers: The biggest difference in this code is that it previously rendered a flexbox layout with no styling, and it is now styled. The styling here was copied from the XY visualization.

Reply:
What styling specifically?